homeassistant/components/buienradar/sensor.py

```diff
@@ -722,13 +722,13 @@ async def async_setup_entry(
         for description in SENSOR_TYPES
     ]
 
+    async_add_entities(entities)
```

Comment:
This won't solve the problem. `async_add_entities` schedules the work to add the entities. They are not added when the call returns.

The correct solution would be to refactor the integration to use a DataUpdateCoordinator.

We have some workarounds already in the `data_updated` method that checks for `hass` etc. Another check could perhaps be added there. Although at some point I think we should just say stop, and disable the integration if it's not maintained and causes problems. This integration doesn't even use a 3rd party library for the requests.

Reply:
Since `async_add_entities` uses `eager_start=True` the task is executed right away. When the entities get the update they are already added. Am I not interpreting right? I've also tested it and the issue was solved.

Nevertheless, this is probably too fragile and can easily break if `async_add_entities` internal behaviour changes, so I'll look into your suggestion on checking it on `data_updated`

Reply:
That code path is just an optimization. We can't rely on that.